core/spec/models/spree/order_contents_spec.rb

```diff
@@ -54,6 +54,19 @@
       expect(order.total.to_f).to eq(19.99)
     end
 
+    context 'when store_credits payment' do
+      let!(:payment) { create(:store_credit_payment, order: order) }
+
+      it 'is expected to have store_credit_payment' do
+        expect(order.payments.store_credits).to include(payment)
```

Comment:
I think this spec is unnecessary because you created this spec in the `before` block above. If you wish to test if the count of payments changed, it would be neater to use RSpec [change matcher](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher) :)